vortex-serde/src/io/object_store.rs

```diff
@@ -68,29 +68,38 @@ impl ObjectStoreReadAt {
 }
 
 impl VortexReadAt for ObjectStoreReadAt {
-    async fn read_at_into(&self, pos: u64, mut buffer: BytesMut) -> io::Result<BytesMut> {
-        let start_range = pos as usize;
-        let bytes = self
-            .object_store
-            .get_range(&self.location, start_range..(start_range + buffer.len()))
-            .await?;
-        buffer.as_mut().copy_from_slice(bytes.as_ref());
-        Ok(buffer)
+    fn read_at_into(
+        &self,
+        pos: u64,
+        mut buffer: BytesMut,
+    ) -> impl Future<Output = io::Result<BytesMut>> + 'static {
+        let object_store = self.object_store.clone();
+        let location = self.location.clone();
+
+        Box::pin(async move {
```

Comment:
i don't think there's a better way of doing this dumb thing until something like https://rust-lang.github.io/rust-project-goals/2024h2/ergonomic-rc.html is real